[docs] Add note in the Contributing guide about linking issues to a PR (#32174)
mui/material-ui: CONTRIBUTING.md

If you have missed a step, don't worry, the Continuous Integration will run a thorough test on your commits and the maintainers of the project can assist.

> Lastly, if your pull request is for resolving a certain problem for which there is an open issue, make sure to link the PR to that issue.

Reviewer: This could be trimmed down, but it's a minor thing.

```suggestion
If your pull request addresses an open issue, make sure to link the PR to that issue.
```

> Additionally, if your pull request is aimed towards resolving a certain problem for which there's an open issue, make sure to link the PR to that issue.

Author: ```suggestion
Lastly, if your pull request is for resolving a certain problem for which there is an open issue, make sure to link the PR to that issue.
```

> You can do that by using any [supported GitHub keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) in the PR description.

Author: ```suggestion
Use any [supported GitHub keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) in the PR description to automatically link them.
```

> Not only this makes it easier to understand the motivation behind the PR as it also automatically closes the issue once the PR is merged.

Author: ```suggestion
This makes it easier to understand where the PR is coming from and also speeds things up as the issue gets closed when the PR is merged.
```
